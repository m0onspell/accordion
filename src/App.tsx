import React from 'react';
import logo from './logo.svg';
import { Settings } from './features/settings';
import { Profile } from './features/profile';
import { Activities } from './features/activities';
import styles from './App.module.css';
import { Switch, Link, Route } from 'react-router-dom';

import { AccordionMenu, MenuItemProps } from './common/components/AccordionMenu';

const navItems: MenuItemProps[] = [
  {
    label: 'Activities',
    href: '/activities',
    items: [
      { label: 'Recreational', href: '/recreational' },
      {
        label: 'Cooking',
        href: '/cooking',
      },
    ],
  },
  {
    label: 'Account',
    href: '/account',
    items: [
      { label: 'Profile', href: '/profile' },
      { label: 'Settings', href: '/settings' },
    ],
  },
];

function App() {
  return (
    <div className={styles.appContainer}>
      <header className={styles.appHeader}>
        <Link to="/" className={styles.appLink}>
          <img src={logo} className={styles.appLogo} alt="logo" />
          <h1>Accordion App</h1>
        </Link>
      </header>

      <AccordionMenu className={styles.sideBar} items={navItems} />

      <main className={styles.main}>
        <Switch>
          <Route path="/account/settings" component={Settings} />
          <Route path="/account/profile" component={Profile} />
          <Route path="/activities/:type" component={Activities} />
          <Route path="/">
            <h1 className={styles.greetings}>Hello!</h1>
          </Route>
        </Switch>
      </main>
    </div>
  );
}

export default App;
