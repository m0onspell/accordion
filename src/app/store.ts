import { configureStore } from '@reduxjs/toolkit';
import settingsSlice from '../features/settings/settingsSlice';
import activitiesSlice from '../features/activities/activitiesSlice';

export const makeStore = () =>
  configureStore({
    reducer: {
      settings: settingsSlice,
      activities: activitiesSlice,
    },
  });

export const store = makeStore();

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type Store = typeof store;
