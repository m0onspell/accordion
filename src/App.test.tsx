import React from 'react';
import { Provider } from 'react-redux';
import { makeStore, Store } from './app/store';
import App from './App';
import { MemoryRouter } from 'react-router';
import * as H from 'history';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

jest.mock('./features/settings/Settings');
jest.mock('./features/profile/Profile');
jest.mock('./features/activities/Activities');
jest.mock('./common/components/AccordionMenu/AccordionMenu');

const setup = ({
  initialEntries,
  store = makeStore(),
}: { initialEntries?: H.LocationDescriptor[]; store?: Store } = {}) => (
  <Provider store={store}>
    <MemoryRouter initialEntries={initialEntries}>
      <App />
    </MemoryRouter>
  </Provider>
);

test('renders correctly', () => {
  const tree = renderer.create(setup({ initialEntries: ['/'] }));
  expect(tree.toJSON()).toMatchSnapshot();
});

test.each`
  initialEntries             | page
  ${['/account/settings']}   | ${'Settings'}
  ${['/account/profile']}    | ${'Profile'}
  ${['/activities/cooking']} | ${'Activities'}
  ${['/']}                   | ${'Hello!'}
`('renders $page', ({ initialEntries, page }) => {
  render(setup({ initialEntries }));
  expect(screen.getByText(page)).toBeInTheDocument();
});
