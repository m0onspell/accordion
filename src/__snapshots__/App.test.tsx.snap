// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="appContainer"
>
  <header
    className="appHeader"
  >
    <a
      className="appLink"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="appLogo"
        src="logo.svg"
      />
      <h1>
        Accordion App
      </h1>
    </a>
  </header>
  AccordionMenu
  <main
    className="main"
  >
    <h1
      className="greetings"
    >
      Hello!
    </h1>
  </main>
</div>
`;
